tests/python_package_test/test_basic.py

```diff
@@ -8,7 +8,7 @@
 from sklearn.model_selection import train_test_split
 
 import lightgbm as lgb
-from lightgbm.compat import PANDAS_INSTALLED, pd_Series
+from lightgbm.compat import PANDAS_INSTALLED, pd_DataFrame, pd_Series
 
 from .utils import load_breast_cancer
```

Comment:
Should we reshape `init_score` for multiclass task before returning it?
https://github.com/microsoft/LightGBM/blob/237ac299fcffa2b6ed9c68917dfdf0767e88e59c/python-package/lightgbm/basic.py#L1927

Reply:
Is there a way to know the number of classes when this method is called?

Reply:
At Dataset level we don't number of classes. But we can get number of data points
https://lightgbm.readthedocs.io/en/latest/C-API.html#c.LGBM_DatasetGetNumData. I guess for non-multiclass task that number should be equal to `len(self.init_score)`. Number of classes can be gotten by dividing `len(self.init_score)` to #rows of Dataset.

Reply:
Done in ad4495942d486435a7bfa2a5e382dc661c2e6f7a. I put the logic in `get_field` instead of `get_init_score` following the discussion in https://github.com/microsoft/LightGBM/pull/4150#discussion_r666954862, let me know what you think.